driver/main.c

```diff
@@ -209,6 +214,7 @@ static const struct file_operations g_ppm_fops = {
 LIST_HEAD(g_consumer_list);
 static DEFINE_MUTEX(g_consumer_mutex);
 static u32 g_tracepoints_attached; // list of attached tracepoints; bitmask using ppm_tp.h enum
+static u32 g_tracepoints_refs[TP_VAL_MAX];
```

Comment:
We will keep refs for each tracepoint, where each ref is a mask of set bits for each consumer ID requiring the tracepoint.
For example:
```
// consumer 0
consumer->tracepoints_attached = 00000001

// consumer 1
consumer->tracepoints_attached = 00000010

g_tracepoints_refs[0 (SYS_ENTER)] -> 1 << 0 (first consumer)
g_tracepoints_refs[1 (SYS_EXIT)] -> 1 << 1 (second consumer)
```

Reply:
BTW the previous number of `max_consumers` is
```c
static unsigned int max_consumers = 5;
```
Don't know the reason for this number but should we bump it to `8`?

Reply:
Refs are for-tracepoint, not for-consumers.

Reply:
Yeah sorry silly me :man_facepalming: not `8` but `32` since our mask is on `32` bits we can have at max `32` consumers, is it true?

Reply:
let's condider this example

```c
g_tracepoints_refs[SYS_ENTER] = 0x80000001
```
This means consumer 32 and consumer 1 are using the `SYS_ENTER` tracepoint

Reply:
Yep we can have up to 32 consumers; i didn't wanted to use an uint8_t because it was too strict; anyway, we just use first `num_consumers` bits.

Reply:
I think that bumping `max_consumers` above 32 would break other things too, just a guess though.

Reply:
@gnosek do you remember why we have `5` as the `max_consumers` value?